internal/app/singularity/cache_list_linux.go

```diff
@@ -31,82 +34,98 @@ func findSize(size int64) (string, error) {
 	return "", fmt.Errorf("failed to detect file size")
 }
 
-func listLibraryCache() error {
-	// loop through library cache
+// listLibraryCache will loop throught and list all library cache (~/.singularity/cache/library).
+// will return the amount of library containers, the total space thoughts containers are using,
+// and an error if one occures.
+func listLibraryCache(listFiles bool) (int, int64, error) {
+	var totalSize int64
+	count := 0
+
 	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
 	if err != nil {
-		return fmt.Errorf("unable to open library cache folder: %v", err)
+		return 0, 0, fmt.Errorf("unable to open library cache folder: %v", err)
 	}
 	for _, f := range libraryCacheFiles {
 		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
 		if err != nil {
-			return fmt.Errorf("unable to look in library cache: %v", err)
+			return 0, 0, fmt.Errorf("unable to look in library cache: %v", err)
 		}
 		for _, c := range cont {
 			fileInfo, err := os.Stat(filepath.Join(cache.Library(), f.Name(), c.Name()))
 			if err != nil {
-				return fmt.Errorf("unable to get stat for library cache: %v", err)
+				return 0, 0, fmt.Errorf("unable to get stat for library cache: %v", err)
 			}
 			printFileSize, err := findSize(fileInfo.Size())
 			if err != nil {
 				// no need to describe the error, since it is already
 				sylog.Warningf("%v", err)
 			}
-			fmt.Printf("%-22s %-22s %-16s %s\n", c.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "library")
+			if !listFiles {
```

Comment:
the name `listFiles` suggests that if it's true, then will list the files. The code is doing the opposite. Maybe call the argument `onlyDirectories`? Or reverse the sense of the argument, so that if it's true, then the file information is listed?